src/Sulu/Component/Content/Types/ResourceLocator/Mapper/PhpcrMapper.php

```diff
@@ -276,10 +276,15 @@ public function loadByResourceLocator($resourceLocator, $webspaceKey, $languageC
                 /** @var NodeInterface $realPath */
                 $realPath = $route->getPropertyValue('sulu:content');
 
-                throw new ResourceLocatorMovedException(
-                    $this->getResourceLocator($realPath->getPath(), $webspaceKey, $languageCode, $segmentKey),
-                    $realPath->getIdentifier()
-                );
+                $locator = $this->getResourceLocator($realPath->getPath(), $webspaceKey, $languageCode, $segmentKey);
+                if (false === $locator) {
+                    throw new ResourceLocatorNotFoundException(\sprintf(
+                        'Couldn\'t generate resource locator for path: %s',
+                        $realPath->getPath()
+                    ));
```

Comment:
@wachterjohannes are you fine with this change?

Reply:
Before it would throw a type error, so it's a better error message.

Reply:
@mamazu @alexander-schranz yes it makes sense to have that